src/utils/isStandardSyntaxUrl.js

```diff
@@ -1,21 +1,35 @@
-import { hasInterpolation } from "../utils"
+import {
+  hasLessInterpolation,
+  hasScssInterpolation,
+} from "../utils"
 /**
  * Check whether a URL is standard
  *
  * @param {string} url
  * @return {boolean} If `true`, the url is standard
  */
 export default function (url) {
-  // SCSS or Less interpolation
-  if (hasInterpolation(url)) { return false }
+  if (url.length === 0) { return true }
 
-  // Inside `'` and `"` work only interpolation
+  // Sass interpolation works anywhere
+  if (hasScssInterpolation(url)) { return false }
+
+  // Inside `'` and `"` work only LESS interpolation
   if (url[0] === "'" && url[url.length - 1] === "'"
     || url[0] === "\"" && url[url.length - 1] === "\""
-  ) { return true }
+  ) {
+    if (hasLessInterpolation(url)) { return false }
+
+    return true
+  }
+
+  // Less variable works only at the beginning
+  // Check is less variable, allow use '@url/some/path'
+  // https://github.com/less/less.js/blob/3.x/lib/less/parser/parser.js#L547
+  if (url[0] === "@" && /^@@?[\w-]+$/.test(url)) { return false }
 
-  // Sass and Less variables at the beginning or after a + sign within
-  if (url[0] === "$" || url[0] === "@" || /['"]\s*\+\s*[\$@]/.test(url)) { return false }
+  // Sass variables works anywhere
+  if (url.indexOf("$") !== -1) { return false }
```

Comment:
If you need to modify the existing regexp to allow for operators other than `+`  you should do that, instead, because this `indexOf` check is inaccurate.